java/query/builder/Sortable.java

```diff
@@ -67,43 +73,49 @@ default S sort(List<UnboundVariable> vars, TypeQLArg.Order order) {
 
     class Sorting {
 
-        private final List<UnboundVariable> vars;
-        private final TypeQLArg.Order order;
         private final int hash;
+        private final List<UnboundVariable> variables;
+        private final Map<UnboundVariable, TypeQLArg.Order> orders;
```

Comment:
once the list of pairs is received, we normalise it into a `List` of variables and a `Map` of orders